supportApp/videoCompressionSrc/codec_tools.c

```diff
@@ -0,0 +1,222 @@
+/* Streaming video compression tools for libavcodec */
+/* Author: B.A. Sobhani <sobhaniba@ornl.gov> */
+
+#include <libavcodec/avcodec.h>
+#include <libavutil/imgutils.h>
+#include <libavutil/opt.h>
+#include "streaming_vc.h"
+//#include "pgm.h"
+#include "codec_tools.h"
+
+AVCodecContext *c_c = 0; 
+int first = 1;
+AVCodecContext *c_d = 0; 
+int count_d = 0;
+
+
+const AVCodec *codec;
+
+AVCodecContext* init_decoder_context(){
+	AVCodecContext *c= NULL;
+	const AVCodec *codec;
+	avcodec_register_all();
+	codec = avcodec_find_decoder(AV_CODEC_ID_H264);
+	c = avcodec_alloc_context3(codec);
+	avcodec_open2(c, codec, NULL);
+	return c;
+}
+
+AVCodecContext* init_encoder_context(int width, int height){
+	AVCodecContext *c= NULL;
+	avcodec_register_all();
+	codec = avcodec_find_encoder(AV_CODEC_ID_H264);
+	c = avcodec_alloc_context3(codec);
+	c->bit_rate = 400000;
+	c->width = width;
+	c->height = height;
+	//c->qmin = 1;
+	//c->qmax = 20;
+	c->time_base = (AVRational){1, 25};
+	c->framerate = (AVRational){25, 1};
+	//av_opt_set(c->priv_data, "crf", "1", AV_OPT_SEARCH_CHILDREN);
+	//av_opt_set(c->priv_data, "crf", "0", 0);
+	//av_opt_set(c->priv_data, "crf", "20", 0);
+	c->gop_size = 100;
```

Comment:
Some of these should be configurable by the users as well, no?

Reply:
I feel like `bit_rate`, `time_base` and `framerate` at least should be :thinking: